### PR TITLE
Index Genopets habitats

### PR DESCRIPTION
**Pull request by ray-kast**
- Add stub Genopets dependency.
- Added stub ingest logic for Genopets.
- Added partial GraphQL interface for Genopets.

### CODE REVIEW

File: crates/core/migrations/2022-07-14-162037_create_geno_habitat_datas/up.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  rent_token          varchar(48) not null,
+  rent_token_decimals smallint    not null,
+  last_rent_payment   timestamp   not null,
+  next_payment_due    timestamp   not null,
+  grace_period        bigint      not null,
+  open_market         bool        not null,
+  slot                bigint      not null,
+  write_version       bigint      not null,
+
+  foreign key (habitat_address) references geno_habitat_datas (address)
```
Comment (imabdulbasit) [inline crates/core/migrations/2022-07-14-162037_create_geno_habitat_datas/up.sql:47]: Shouldn't this be removed if accounts are received in random order

File: crates/genostub/src/state.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    pub total_ki_harvested: u64,
+    pub total_crystals_refined: u64,
+    pub terraforming_habitat: Option<Pubkey>,
+    pub active: bool,
+    pub durability: u16,
+    pub habitats_terraformed: u32,
+    pub sequence: u64,
+    pub guild: Option<u16>,
+    pub sub_habitat_cooldown_timestamp: u64,
+}
```
Comment (mralbertchen) [inline crates/genostub/src/state.rs:55]: We added one last property a few days ago. Can it be added please?

```rust
 pub harvester_settings_cooldown_timestamp: u64
```

File: crates/genostub/src/state.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    pub grace_period: u64,
+    pub open_market: bool,
+}
+
+#[account]
+#[derive(Default)]
+pub struct HabitatData {
+    pub habitat_mint: Pubkey,
+    pub level: u8,
+    pub element: u8,
```
Comment (mralbertchen) [inline crates/genostub/src/state.rs:31]: Did you mean to use `Element` here since you added the enum?

**Review by imabdulbasit — CHANGES_REQUESTED** [review]

**Review by imabdulbasit — APPROVED** [review]

**Review by mpwsh — APPROVED** [review]

**Review by mralbertchen — APPROVED** [review]
LGTM

**Pull request merged**